airgo/refresh_data.go

```diff
@@ -0,0 +1,98 @@
+package airgo
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	airtable "github.com/mottaquikarim/go-airtable"
+	log "github.com/sirupsen/logrus"
+)
+
+type RefreshOpts struct {
+	ApiKey string
+	BaseId string
+	Conf   string
+	Dest   string
+}
+
+type Config struct {
+	Nav []ConfigItem
+}
+
+type ConfigItem struct {
+	Name     string `json:"name"`
+	Key      string `json:"key"`
+	Filename string `json:"filenameRoot"`
+	Viewname string `json:"viewname"`
+}
+
+func RefreshData(cfg RefreshOpts) error {
+	var site []byte
+	var err error
+
+	if len(cfg.Conf) == 0 {
+		return fmt.Errorf("Path to conf cannot be empty")
+	} else if len(cfg.ApiKey) == 0 {
+		return fmt.Errorf("ApiKey must be provided")
+	} else if len(cfg.BaseId) == 0 {
+		return fmt.Errorf("BaseId must be provided")
+	} else if len(cfg.Dest) == 0 {
+		return fmt.Errorf("Path to dest cannot be empty")
+	}
+
+	// read file
+	site, err = ioutil.ReadFile(cfg.Conf)
+	if err != nil {
+		return fmt.Errorf("Failed to read config file %v", err)
+	}
+
+	// parse json content
+	var configData Config
+	if err = json.Unmarshal(site, &configData); err != nil {
+		return fmt.Errorf("Config is invalid %v", err)
+	}
+
+	// if here, we have a config file to process
+	// we create an account struct
+	// that is passed in to airtable
+	// module
+	account := airtable.Account{
+		ApiKey: cfg.ApiKey,
+		BaseId: cfg.BaseId,
+	}
+
+	for _, configItem := range configData.Nav {
```

Comment:
from lines 66 to 69 we access the Airtable and make sure we take into account which table we are using.